hack/manifest-gen/internal/manifest.go

```diff
@@ -72,6 +73,8 @@ func Generate(opts *GenerateFlags) error {
 	client.ReleaseName = "elastic-operator"
 	client.Namespace = opts.OperatorNamespace
 	client.PostRenderer = helmLabelRemover{}
+	fakeKubeVersion, _ := chartutil.ParseKubeVersion("v9.99.99")
+	client.KubeVersion = fakeKubeVersion
```

Comment:
Otherwise the version used is `1.20` coming from the default capacities which makes the generation fail if we want to start to require `>=1.21.0`.

Reply:
[IIUC](https://github.com/helm/helm/blob/b5e39c15ea4f2ac14b1e5b6635d46410ffb79fec/pkg/action/install.go#L216-L222) since Helm is not connected to a real K8S cluster (`ClientOnly == true`) then a K8S API version must be arbitrarily selected. This version is hardcoded in Helm, and as of today its value is `1.20`, which prevents Helm to render the manifests from the Charts if we request a higher value as it is the case in this PR (`1.21`).
I would maybe add a comment to explain the problem and why we are setting this value.